internals/cli/cli.go

```diff
@@ -269,6 +266,13 @@ func (e *exitStatus) Error() string {
 }
 
 func Run() error {
+	// the configuration of the Client used by all commands.
+	var clientConfig client.Config
+	_, clientConfig.Socket = getEnvPaths()
+	return RunWithConfig(&clientConfig)
+}
+
+func RunWithConfig(config *client.Config) error {
```

Comment:
Could it be useful, given the context of ```cli```, to introduce a new cli options struct ?

```go
type Options struct {
    client    *client.Config
}

func RunWithConfig(config *Config) error {}
```

This would, for example, allow us to then add options such as logger output in the future (```logger.SetLogger```).

Reply:
We could have two runs, but its not breaking an external API if we change the one I think ?

Reply:
> We could have two runs, but its not breaking an external API if we change the one I think ? Its only called once in main.go I believe.

Yes, a single `Run()` that always takes a `*client.Config` would also be possible. Depending on what @benhoyt prefers, I can update the PR.

Reply:
Yes, I agree a new options struct would be good, to allow for easy additions later. But maybe we call it `RunOptions`. And I think we can change the signature of `Run` here (seeing it's "internals") instead of adding a new function:

```go
type RunOptions struct {
    ClientConfig *client.Config
}

func Run(options *RunOptions) error { ... }
```

I think it should be okay `options` to be nil or for `ClientConfig` to be nil, in which case it uses the current defaults. So `main.go` will probably just call it as `cli.Run(nil)`.

Reply:
It's tempting to add a `Logger logger.Logger` option while we're here (default if nil would be the current version). Yeah, let's do that. It's simple and an obvious dependency, so makes sense for it to be in the options struct.

Reply:
Done :)